### ISSUE TITLE
Use multiple CPUs or distribute on cluster

### ISSUE THREAD

**Issue by kripper** [Lightning-AI/lit-gpt#344]
I'm trying the lit-gpt CPU version on a 32 core server.
I noticed that only one CPU is used.
I tried adding `torch.set_num_threads(num_threads)` but it still uses only 1 CPU.

How can we use multiple CPUs or distribute computing on multiple servers?

**Comment by kripper** [comment]
Maybe optimizations are out of the scope of lit-gpt.
I tested the same model with llama.cpp and CTranslate2 and both are 32+ times faster (they use all CPUs and also support distributed computing).

**Comment by carmocca** [comment]
I agree with you. PyTorch (what Lit-GPT uses) is not competitive with bare metal projects like those for inference on CPUs.
